homework07/task_01.py

```diff
@@ -0,0 +1,194 @@
+# Once created record can't be edited, only removed.
+# Contact names can't duplicate.
+
+# Displays quantity of existing records in phonebook
+def show_stats(phonebook: dict):
+    if not phonebook:
+        print("= PHONEBOOK IS EMPTY =")
+    else:
+        print(f"= You have {len(phonebook)} contacts =")
+    return phonebook
+
+
+# Confirms if name is valid to be recorded or rejects it
+def check_available(name: str, contacts):
+    if not name or name.isspace() or name in contacts:
+        print("= Record NOT created. Reason :", end=" ")
+        if not name:
+            print("Entered nothing =")
+        elif name.isspace():
+            print("Can't accept only spaces =")
+        else:
+            print("Contact already exists =")
+        return False
+    return True
+
+
+# Adds new contact record to phonebook
+def add_contact(phonebook: dict):
+    contact_data = {"Phone number": "",
+                    "E-mail": "",
+                    "Address": ""
+                    }
+    print("Enter contact name")
+    name = input("Name: -> ")
+
+    if check_available(name, phonebook.keys()):
+        print(f"Enter < {name} > contact data")
+
+        for info in contact_data.keys():
+            contact_data[info] = input(f"{info}: > ")
+
+        phonebook[name] = contact_data
+        print(f"Contact < {name} > created.")
+
+    return phonebook
+
+
+# Removes contact record from phonebook
+def delete_contact(name: str, phonebook: dict):
+    if name in phonebook.keys():
+        if input(f"Type 'yes' if you agree to delete < {name} >: -> ").lower() == "yes":
+            phonebook.pop(name)
+            print(f"= Contact < {name} > deleted =")
+    else:
+        print(f"= Contact < {name} > doesn't exist =")
+    return phonebook
+
+
+# Displays all contacts which have any matches with searched name
+def search(match: str, phonebook: dict):
+    found = []
+    for name in phonebook.keys():
+        if match.lower() in name.lower():
+            found.append(name)
+    if not found:
+        print(f"< {match} > not found")
+    else:
+        print("Matches found in:")
+        for finding in found:
+            print(f"< {finding} >")
+    return phonebook
+
+
+# Displays all recorded names in phonebook
+def list_names(phonebook: dict):
+    if not phonebook.keys():
+        print("= There is no contacts yet =")
+    else:
+        print("Recorded contacts:")
+        for contact in phonebook.keys():
+            print(f" < {contact} >")
+    return phonebook
+
+
+# Displays selected contact info
+def show_name(name: str, phonebook: dict):
+    if name not in phonebook.keys():
+        print(f"= There is no contact < {name} > =")
+    else:
+        print(f"< {name} >")
+        for info, data in phonebook[name].items():
+            print(f" - {info}: {data}")
+    return phonebook
+
+
+# Extracts name parameter from user's query
+def extract_name(entry: str):
```

Comment:
nit: add comments to function inside a function and describe how it works. Especially when there is something not clear like `len(entry) >= 3 and entry[0] == "<" and entry[-1] == ">"`
```suggestion
def extract_name(entry: str):
    """
    Extracts name parameter from user's query
    
    ....how it works
    """
```